mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb

```diff
@@ -0,0 +1,136 @@
+  provider = google-beta
+  pool = google_privateca_ca_pool.default.name
+  certificate_authority_id = "<%= ctx[:vars]['privateca_certificate_authority_id'] %>"
+  location = "us-central1"
+  lifetime = "86400s"
+  type = "SELF_SIGNED"
+  deletion_protection = false
+  skip_grace_period = true
+  ignore_active_certificates_on_deletion = true
+  config {
+    subject_config {
+      subject {
+        organization = "Test LLC"
+        common_name = "my-ca"
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = true
+      }
+      key_usage {
+        base_key_usage {
+          cert_sign = true
+          crl_sign = true
+        }
+        extended_key_usage {
+          server_auth = false
+        }
+      }
+    }
+  }
+  key_spec {
+    algorithm = "RSA_PKCS1_4096_SHA256"
+  }
+}
+
+resource "google_project_service_identity" "ns_sa" {
+  provider = google-beta
+
+  service = "networksecurity.googleapis.com"
+}
+
+resource "google_privateca_ca_pool_iam_member" "tls_inspection_permission" {
+  provider = google-beta
+
+  ca_pool = google_privateca_ca_pool.default.id
+  role = "roles/privateca.certificateManager"
+  member = "serviceAccount:${google_project_service_identity.ns_sa.email}"
+}
+
+resource "google_certificate_manager_trust_config" "default" {
+  provider    = google-beta
+  name        = "<%= ctx[:vars]['trust_config_name'] %>"
+  description = "sample trust config description"
+  location    = "us-central1"
+
+  trust_stores {
+    trust_anchors {
+      pem_certificate = file("test-fixtures/ca_cert.pem")
+    }
+    intermediate_cas {
+      pem_certificate = file("test-fixtures/ca_cert.pem")
+    }
+  }
+
+  labels = {
+    foo = "bar"
+  }
+}
+
+resource "google_network_security_tls_inspection_policy" "<%= ctx[:primary_resource_id] %>" {
+  provider = google-beta
+  name     = "<%= ctx[:vars]['resource_name'] %>"
+  location = "us-central1"
+  ca_pool  = google_privateca_ca_pool.default.id
+  exclude_public_ca_set = false
+
+  min_tls_version     = "TLS_1_0"
+  trust_config        = "projects/${data.google_project.project.number}/locations/${google_certificate_manager_trust_config.default.location}/trustConfigs/${google_certificate_manager_trust_config.default.name}"
+  tls_feature_profile = "PROFILE_CUSTOM"
+  custom_tls_features = [
+    "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
+    "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+    "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
+    "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+    "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+    "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+    "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+    "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+    "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+    "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+    "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
+    "TLS_RSA_WITH_AES_128_CBC_SHA",
+    "TLS_RSA_WITH_AES_128_GCM_SHA256",
+    "TLS_RSA_WITH_AES_256_CBC_SHA",
+    "TLS_RSA_WITH_AES_256_GCM_SHA384",
+  ]
+
+  depends_on = [
```

Comment:
All these shouldn't be needed anymore if you use the implicit dependencies recommended above.

Reply:
I put those just in case of the replay test

Reply:
@maxi-cit not sure what you mean. I'll let @NickElliot elaborate on this.
For sure it's not something I see super often in our code. I'd avoid it as much as possible.

Reply:
@LucaPrete there are times where you need to be explicit about the implicit specially when you create similar resources. And this is due some REPLAYING check where resources should be deleted the opposite way they were created. At least, it is what I understand.